modules/core/shared/src/main/scala/lucuma/core/model/sequence/gmos/binning.scala

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model.sequence.gmos
+
+import cats.Order
+import cats.syntax.all.*
+import coulomb.Quantity
+import eu.timepit.refined.types.numeric.PosDouble
+import eu.timepit.refined.types.numeric.PosInt
+import lucuma.core.enums.GmosNorthDetector
+import lucuma.core.enums.GmosNorthFpu
+import lucuma.core.enums.GmosNorthGrating
+import lucuma.core.enums.GmosSouthDetector
+import lucuma.core.enums.GmosSouthFpu
+import lucuma.core.enums.GmosSouthGrating
+import lucuma.core.enums.GmosXBinning
+import lucuma.core.enums.GmosYBinning
+import lucuma.core.enums.ImageQuality
+import lucuma.core.math.Angle
+import lucuma.core.math.Wavelength
+import lucuma.core.math.units.NanometersPerPixel
+import lucuma.core.model.SourceProfile
+import spire.math.Rational
+
+import java.math.RoundingMode.HALF_UP
+
+/**
+ * Optimal GMOS spectral and spatial binning calculations.
+ *
+ * See https://docs.google.com/document/d/1P8_pXLRVomUSvofyVkAniOyGROcAtiJ7EMYt9wWXB0o.
+ */
+object binning {
+
+  val DefaultSampling: PosDouble =
+    PosDouble.unsafeFrom(2.5)
+
+  /**
+   * Object angular size estimate based on source profile and image quality.
+   */
+  def objectSize(p: SourceProfile, iq: ImageQuality): Angle =
+    p match {
+      case SourceProfile.Point(_)          => iq.toAngle
+      case SourceProfile.Uniform(_)        => Angle.Angle180
```

Comment:
The fact that the spatial binning calculation ignores the slit width (is that intended?) means that a `Uniform` source profile will always result in a y-binning of 4.  Is that expected?

Reply:
"Any" large value is fine.  I think basically it is saying that a `Uniform` source does not have a meaningful limiting size.